estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -665,12 +673,14 @@ async def _binding_incremental_task(
 
             if lag > binding.resourceConfig.interval:
                 # We're not idle. Attempt to fetch the next changes.
+                sleep_for = timedelta()
                 continue
             else:
                 # We're idle. Sleep until the cursor is `interval` old.
                 sleep_for = binding.resourceConfig.interval - lag
         else:
             # We're not idle. Attempt to fetch the next changes.
+            sleep_for = timedelta()
             continue
 
         task.log.debug("incremental task is idle", {"sleep_for": sleep_for, "cursor": state.cursor})
```

Comment:
You have to move the below sleep up to the top, or you're sleeping twice.

Reply:
🤦 I had that and it got lost when rebasing, thanks. Updated